datagen/src/main/scala/com/nvidia/rapids/tests/scaletest/TableGenerator.scala

```diff
@@ -0,0 +1,231 @@
+package com.nvidia.rapids.tests.scaletest
+
+import org.apache.spark.sql.{DataFrame, SparkSession}
+import org.apache.spark.sql.tests.datagen.{DBGen, ExponentialDistribution, FlatDistribution, MultiDistribution}
+
+import scala.util.Random
+
+/**
+ * pre-concepts:
+ * - key groups:
+ * 1. The primary key for table a. In table a each row will be unique
+ * 2. Adjusted key group. The number of columns will correspond to the complexity. The types of the columns should be selected from the set {string, decimal(7, 2), decimal(15, 2), int, long, timestamp, date, struct<num: long, desc: string>}
+ * 3. 3-column key group with the types string, date, long.
+ * 4. 1 column that is an int with a unique count of 5
+ */
+class TableGenerator(scaleFactor: Int, complexity: Int, spark: SparkSession) {
+  val dbgen: DBGen = DBGen()
+  // define numRows in class as different table may depend on each other
+  private val aNumRows = scaleFactor * 10000
+  private val bNumRows = scaleFactor * 1000000
+  private val cNumRows = scaleFactor * 100000
+  private val dNumRows = scaleFactor * 100000
+  private val eNumRows = scaleFactor * 1000000
+  private val fNumRows = scaleFactor * 10000
+  private val gNumRows = scaleFactor * 1000000
+  private val seed = 41
+  private val random = new Random(seed)
+
+  private def randomColumnType(): String = {
+    // TODO: add more types when they are supported e.g. BinaryType CalendarintervalType etc.
+    // See: https://github.com/NVIDIA/spark-rapids/issues/8717
+    // ride-along/data columns will pick the column type from the candidates below.
+    // required at https://github.com/NVIDIA/spark-rapids/issues/8813#issue-1822958165
+    // TODO: maybe we can move this to an external text file so we can extend it easily in the future
+    val candidates = Seq(
+      "int",
+      "long",
+      "Decimal(7, 2)",
+      "Decimal(19, 4)",
+      "string",
+      "map<string, struct<lat: Decimal(7,4), lon: Decimal(7, 4)>>",
+      "timestamp",
+      "date")
+    candidates(random.nextInt(candidates.length))
+  }
+
+  private def randomNumericColumnType(): String = {
+    val candidates = Seq(
+      "int",
+      "long",
+      "Decimal(7, 2)",
+      "Decimal(19, 4)"
+    )
+    val seed = 41
+    val r = new Random(seed)
```

Comment:
Isn't this always going to return the same type? Should it use the random above instead?